inc/azure_uhttp_c/uhttp.h

```diff
@@ -33,7 +33,7 @@ typedef struct HTTP_CLIENT_HANDLE_DATA_TAG* HTTP_CLIENT_HANDLE;
 /** @brief Enumeration specifying the possible return values for the APIs in
 *           this module.
 */
-MU_DEFINE_ENUM(HTTP_CLIENT_RESULT, HTTP_CLIENT_RESULT_VALUES);
+MU_DEFINE_ENUM_WITHOUT_INVALID(HTTP_CLIENT_RESULT, HTTP_CLIENT_RESULT_VALUES);
 
```

Comment:
Not clear on how this fixes the problem.  When I looked at the issue, it seems like 
https://github.com/Azure/azure-iot-sdk-c/blob/master/provisioning_client/src/prov_transport_http_client.c#L1234
```
result = uhttp_client_set_option(http_info->http_client, option, value);
```
should be checked against HTTP_CLIENT_OK, and if equivalent, then result set to 0.